src/components/notification-tray/notification-tray.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+} from "@stencil/core"
+
+@Component({
+  tag: "oui-notification-tray",
+  styleUrl: "notification-tray.css",
+})
+export class NotificationTray {
+  @Element() public tray: HTMLElement
+
+  /**
+   * Open or close the notification drawer
+   */
+  @Prop({ reflect: true, mutable: true }) public opened = false
+
+  /**
+   * Direction of the drawer
+   */
+  @Prop() public direction: "to-right" | "to-left" = "to-left"
+
+  /**
+   *
+   */
+  @Prop({ reflect: true, mutable: true }) public count: number = 0
+
+  /**
+   *
+   */
+  @Prop({ reflect: true, mutable: true }) public unread: number = 0
+
+  /**
+   *
+   */
+  @Event() public dismissall: EventEmitter
+  public dismissallHandler = () => {
+    this.dismissall.emit()
+  }
+
+  /**
+   *
+   */
+  @Event() public dismiss: EventEmitter
+  public dismissHandler = (el: HTMLElement) => {
+    this.dismiss.emit(el.dataset.notificationName)
+  }
+
+  @Listen("click")
+  public todoCompletedHandler(evt: UIEvent) {
+    evt.preventDefault()
+    evt.stopPropagation()
+    const target: HTMLElement = evt.target as HTMLElement
```

Comment:
No need to declare the type on both left hand of assignment and through coercion on the right hand side using `as`, typescript is smart enough to infer from the `as`

Reply:
```suggestion
    const target = evt.target as HTMLElement
```

Reply:
done :)